views/calendar/calendar.js

```diff
@@ -68,47 +69,50 @@ export default class CalendarView extends React.Component {
     return `${calendarUrl}?${qs.stringify(params)}`
   }
 
-  getEvents = async () => {
+  getEvents = async (now: moment=moment.tz(TIMEZONE)) => {
     let url = this.buildCalendarUrl(this.props.calendarId)
 
-    let data = null
-    let error = null
+    let data: GoogleCalendarEventType[] = []
     try {
-      let result = await fetch(url).then(r => r.json())
-      error = result.error
+      let result = await fetchJson(url)
+      const error = result.error
+      if (error) {
+        tracker.trackException(error.message)
+        this.setState({error: error})
+      }
+
       data = result.items
     } catch (error) {
       tracker.trackException(error.message)
       this.setState({error: error.message})
-      console.error(error)
+      console.warn(error)
     }
 
-    if (data && data.length) {
-      let now = moment.tz(TIMEZONE)
-      data.forEach(event => {
-        event.startTime = moment(event.start.date || event.start.dateTime)
-        event.endTime = moment(event.end.date || event.end.dateTime)
-        event.isOngoing = event.startTime.isBefore(now, 'day')
-      })
-      let grouped = groupBy(data, event => {
-        if (event.isOngoing) {
-          return 'Ongoing'
-        }
-        let isToday = event.startTime.isSame(now, 'day')
-        if (isToday) {
-          return 'Today'
-        }
-        return event.startTime.format('ddd  MMM Do')  // google returns events in CST
-      })
-      this.setState({events: this.state.events.cloneWithRowsAndSections(grouped)})
-    } else if (data && !data.length) {
-      this.setState({noEvents: true})
-    }
-    if (error) {
-      tracker.trackException(error.message)
-      this.setState({error: error.message})
-    }
-    this.setState({loaded: true})
+    const events: EventType[] = data.map((event: GoogleCalendarEventType) => {
+      const startTime = moment(event.start.date || event.start.dateTime)
+      const endTime = moment(event.end.date || event.end.dateTime)
+      return {
+        ...event,
+        startTime,
+        endTime,
+        isOngoing: startTime.isBefore(now, 'day'),
+      }
+    })
+
+    const grouped = groupBy(events, event => {
```

Comment:
Then I group them in a second step. I also removed at least one level of indentation, by early-exits.